src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -1090,7 +1090,35 @@ def extract_assay_expression(args):
     # assay names listing
     descriptor = DXDataset(dataset_handler.dataset, project=dataset_handler.datasetRecordProject).get_descriptor()
     dataset_handler.populate_descriptor(descriptor)
-    print(dataset_handler.list_assay_names("molecular_expression"))
+    
+    assay_names_list = dataset_handler.get_assay_names("molecular_expression")
+
+    if parser_dict.get("list_assays"):
+        print(assay_names_list)
+        sys.exit(0)
+
+    # possible assay picking
+    if parser_dict.get("assay_name"):
+        assay_name = parser_dict.get("assay_name")
+        # print(assay_name)
+        assay_index = dataset_handler.get_assay_indice_in_list(assay_names_list, assay_name)
```

Comment:
Suggestion for 1101-1106 replacement:

```suggestion
        assay_index = dataset_handler.get_assay_indice_in_list(assay_names_list, args.assay_name) if args.assay_name else 0
```